core/tests/ts-integration/tests/fees.test.ts

```diff
@@ -143,6 +174,116 @@ testFees('Test fees', () => {
         console.log(`Full report: \n\n${reports.join('\n\n')}`);
     });
 
+    test('Test gas price expected value', async () => {
+        const receiver = ethers.Wallet.createRandom().address;
+        const l1GasPrice = 2_000_000_000n; /// set to 2 gwei
+        const baseTokenAddress = await alice._providerL2().getBaseTokenContractAddress();
+        const isETHBasedChain = baseTokenAddress == zksync.utils.ETH_ADDRESS_IN_CONTRACTS;
+
+        await setInternalL1GasPrice(alice._providerL2(), {
+            newL1GasPrice: l1GasPrice.toString(),
+            newPubdataPrice: l1GasPrice.toString(),
+            customBaseToken: !isETHBasedChain
+        });
+
+        const receipt = await (
+            await alice.sendTransaction({
+                to: receiver,
+                value: BigInt(1)
+            })
+        ).wait();
+
+        const expectedETHGasPrice =
+            testMaster.environment().l1BatchCommitDataGeneratorMode === DataAvailabityMode.Rollup
+                ? 100_000_000 // 0.1 gwei (the minimum)
+                : 110_000_000; // 0.11 gwei, in validium we need to add compute overhead
+        const expectedCustomGasPrice = expectedETHGasPrice * 0.314;
+
+        if (isETHBasedChain) {
+            expect(receipt.gasPrice).toBe(BigInt(expectedETHGasPrice));
+        } else {
+            expect(receipt.gasPrice).toBe(BigInt(expectedCustomGasPrice));
+        }
+
+        if (!isETHBasedChain) {
+            await setInternalL1GasPrice(alice._providerL2(), {
+                newL1GasPrice: l1GasPrice.toString(),
+                newPubdataPrice: l1GasPrice.toString(),
+                customBaseToken: true,
+                forcedBaseTokenRatioNumerator: 271,
+                forcedBaseTokenRatioDenominator: 100
+            });
+
+            const receipt2 = await (
+                await alice.sendTransaction({
+                    to: receiver,
+                    value: BigInt(1)
+                })
+            ).wait();
+
+            const expectedCustomGasPrice = expectedETHGasPrice * 2.71;
+            expect(receipt2.gasPrice).toBe(BigInt(expectedCustomGasPrice));
+        }
+    });
+
+    test('Test base token ratio fluctuations', async () => {
+        const l1GasPrice = 2_000_000_000n; /// set to 2 gwei
+        const baseTokenAddress = await alice._providerL2().getBaseTokenContractAddress();
+        const isETHBasedChain = baseTokenAddress == zksync.utils.ETH_ADDRESS_IN_CONTRACTS;
+
+        if (isETHBasedChain) return;
+
+        await setInternalL1GasPrice(alice._providerL2(), {
+            newL1GasPrice: l1GasPrice.toString(),
+            newPubdataPrice: l1GasPrice.toString(),
+            customBaseToken: true,
+            forcedBaseTokenRatioNumerator: 300,
+            forcedBaseTokenRatioDenominator: 100,
+            forcedBaseTokenRatioFluctuationsAmplitude: 20,
```

Comment:
Why not to use the same naming as we have in the forced api client already? I.e. 
 `externalPriceApiClientForcedFluctuation` for example.